src/cli.rs

```diff
@@ -1276,6 +1377,58 @@ impl<'a> BirliContext<'a> {
         Ok(result)
     }
 
+    /// Call `run()` for every channel range in `self.channel_range_sel`.
+    pub fn run_ranges(self) -> Vec<Result<HashMap<String, Duration>, BirliError>> {
+        let ranges = self.channel_range_sel.ranges.clone();
+        let mut ranged_context: BirliContext = BirliContext {
+            corr_ctx: self.corr_ctx,
+            prep_ctx: self.prep_ctx,
+            vis_sel: self.vis_sel,
+            flag_ctx: self.flag_ctx,
+            io_ctx: self.io_ctx,
+            avg_time: self.avg_time,
+            avg_freq: self.avg_freq,
+            num_timesteps_per_chunk: self.num_timesteps_per_chunk,
+            ignore_dut1: self.ignore_dut1,
+            channel_range_sel: self.channel_range_sel,
+        };
+        let original_io_ctx = ranged_context.io_ctx.clone();
+        ranged_context.vis_sel = ranged_context.vis_sel.clone();
+        ranged_context.io_ctx = ranged_context.io_ctx.clone();
+        let mut results: Vec<Result<HashMap<String, Duration>, BirliError>> =
+            Vec::with_capacity(ranges.len());
+        for (range_start, range_end) in ranges {
+            ranged_context.vis_sel.coarse_chan_range = range_start..range_end + 1;
+            //ranged_context.
+            ranged_context.io_ctx = original_io_ctx.clone();
+            ranged_context.io_ctx.ms_out = ranged_context.io_ctx.ms_out.map(|path| {
+                let mut path = path.clone();
+                path.set_file_name(format!(
+                    "{}_{}-{}.ms",
+                    path.file_stem().unwrap().to_str().unwrap(),
```

Comment:
These things are tedious and annoying, but we should try to avoid unwraps. I can help offer my experience, doing this myself